plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -335,6 +346,79 @@ func (r *NodeAuthorizer) authorizeResourceSlice(nodeName string, attrs authorize
 	return r.authorize(nodeName, sliceVertexType, attrs)
 }
 
+// authorizeNode authorizes node requests to Node API objects
+func (r *NodeAuthorizer) authorizeNode(nodeName string, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
+	switch attrs.GetSubresource() {
+	case "":
+		switch attrs.GetVerb() {
+		case "create", "update", "patch":
+			// Use the NodeRestriction admission plugin to limit a node to creating/updating its own API object.
```

Comment:
prior to this change, when node/A attempted to update node/B, didn't this authorizer return false because there was no path from node/A to node/B?  Why change this here for update/patch?

Reply:
> prior to this change, when node/A attempted to update node/B, didn't this authorizer return false because there was no path from node/A to node/B?

No, prior to this change, the node authorizer allowed Node updates unconditionally. Node and Pod resources weren't even in this switch statement:

https://github.com/kubernetes/kubernetes/blob/5fc7032a0e509d41890573d5ea2beec880593bda/plugin/pkg/auth/authorizer/node/node_authorizer.go#L114-L143

so all requests for those resources fell through to the generic RBAC rules for nodes and pods:

https://github.com/kubernetes/kubernetes/blob/f2428d66cc2256fcbe2a486b1403603530f46c6f/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L119-L123

https://github.com/kubernetes/kubernetes/blob/f2428d66cc2256fcbe2a486b1403603530f46c6f/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L128-L139

We completely relied on NodeRestriction admission to guard writes, and just broadly allowed reads.

> Why change this here for update/patch?

This PR translates those permissions from RBAC rules to coded rules, and locks down the reads. It's not changing the existing write permissions. We still rely on NodeRestriction to guard writes.

I added positive and negative unit tests for these resources, adjusted the node authorizer unit tests to run with this feature both enabled and disabled, and ensure the results are identical (except the reads of unrelated objects which are now more locked down).